static/app/components/events/eventStatisticalDetector/lineChart.tsx

```diff
@@ -22,85 +22,98 @@ interface ChartProps {
-      grid: {
-        top: '40px',
-        bottom: '0px',
-      },
-      legend,
-      toolBox: {show: false},
-      tooltip: {
-        valueFormatter: (value, seriesName) => {
-          return tooltipFormatter(value, aggregateOutputType(seriesName));
-        },
-      },
-      xAxis: {type: 'time'},
-      yAxis: {
-        minInterval: durationUnit,
-        axisLabel: {
-          color: theme.chartLabel,
-          formatter: (value: number) =>
-            axisLabelFormatter(value, 'duration', undefined, durationUnit),
+  const durationUnit = getDurationUnit(series);
+
+  const chartOptions: Omit<React.ComponentProps<typeof EChartsLineChart>, 'series'> = {
+    axisPointer: {
+      link: [
+        {
+          xAxisIndex: [0, 1],
+          yAxisIndex: [0, 1],
         },
+      ],
+    },
+    colors: [theme.gray200, theme.gray500],
+    grid: {
+      top: '40px',
+      bottom: '0px',
+    },
+    legend,
+    toolBox: {show: false},
+    tooltip: {
+      valueFormatter: (value, seriesName) => {
+        return tooltipFormatter(value, aggregateOutputType(seriesName));
       },
-      options: {
-        visualMap: VisualMap({
-          show: false,
-          type: 'piecewise',
-          selectedMode: false,
-          dimension: 0,
-          pieces: [
-            {
-              gte: 0,
-              lt: evidenceData?.breakpoint ? evidenceData.breakpoint * 1000 : 0,
-              color: theme.gray500,
-            },
-            {
-              gte: evidenceData?.breakpoint ? evidenceData.breakpoint * 1000 : 0,
-              color: theme.red300,
-            },
-          ],
-        }),
+    },
+    xAxis: {type: 'time'},
+    yAxis: {
+      minInterval: durationUnit,
+      axisLabel: {
+        color: theme.chartLabel,
+        formatter: (value: number) =>
+          axisLabelFormatter(value, 'duration', undefined, durationUnit),
       },
-    };
-  }, [series, theme, percentileSeries, evidenceData.breakpoint]);
+    },
+    options: {
+      visualMap: VisualMap({
+        show: false,
+        type: 'piecewise',
+        selectedMode: false,
+        dimension: 0,
+        pieces: [
+          {
+            gte: 0,
+            lt: evidenceData?.breakpoint ? evidenceData.breakpoint * 1000 : 0,
+            color: theme.gray500,
+          },
+          {
+            gte: evidenceData?.breakpoint ? evidenceData.breakpoint * 1000 : 0,
+            color: theme.red300,
+          },
+        ],
+      }),
+    },
+  };
+  return {
+    series,
+    chartOptions,
+    datetime,
+  };
+}
+
+function LineChart({datetime, percentileSeries, evidenceData}: ChartProps) {
+  const theme = useTheme();
+  const router = useRouter();
+
+  const {series, chartOptions} = useMemo(() => {
+    return getBreakpointChartPropsFromData(
+      {datetime, percentileSeries, evidenceData},
+      theme
+    );
+  }, [percentileSeries, evidenceData, theme, datetime]);
 
   return (
     <ChartZoom router={router} {...datetime}>
```

Comment:
Right now I'm only generating `EChartsLineChart` without the zoomRenderProps. My understanding was that it's needed for dynamic rendering but please correct me if I'm wrong.